http/src/headers/unknown_header_name.rs

```diff
@@ -46,6 +44,30 @@ impl<'a> From<UnknownHeaderName<'a>> for HeaderName<'a> {
     }
 }
 
+impl UnknownHeaderName<'_> {
+    pub fn is_valid(&self) -> bool {
+        self.0
+            .chars()
+            .all(|c| matches!(c, 'a'..='z'|'A'..='Z'|'0'..='9'|'-'|'_'))
```

Comment:
This should at least allow `.`, per https://www.rfc-editor.org/rfc/rfc9110.html#section-16.3.2.1-3 and possibly all of `token`, defined at https://www.rfc-editor.org/rfc/rfc9110.html#name-tokens, because of https://www.rfc-editor.org/rfc/rfc9110.html#section-5.1-2.

Reply:
Thanks! Broadened to all of tchar for correctness